src/librustc_platform_intrinsics/lib.rs

```diff
@@ -32,34 +33,69 @@ pub enum Type {
     Void,
     Integer(/* signed */ bool, u8, /* llvm width */ u8),
     Float(u8),
-    Pointer(Box<Type>, Option<Box<Type>>, /* const */ bool),
-    Vector(Box<Type>, Option<Box<Type>>, u8),
-    Aggregate(bool, Vec<Type>),
+    Pointer(&'static Type, Option<&'static Type>, /* const */ bool),
+    Vector(&'static Type, Option<&'static Type>, u8),
+    Aggregate(bool, &'static &'static [&'static Type]),
```

Comment:
That does seem a little odd, actually. Since it's not going to change the overall size of the type, the alignment means that `Pointer` and `Vector` are 4 words big anyway.